### PR TITLE
Replace black with ruff in `tidy`

### PR DESCRIPTION
**Pull request by Kobzol**
`ruff` can both lint and format Python code (in fact, it should be a mostly drop-in replacement for `black` in terms of formatting), so it's not needed to use `black` anymore. This PR removes `black` and replaces it with `ruff`, to get rid of one Python dependency, and also to make Python formatting faster (although that's a small thing).

If we decide to merge this, we'll need to "reformat the world" - `ruff` is not perfectly compatible with `black`, and it also looks like `black` was actually ignoring some files before. I tried it locally (`./x test tidy --extra-checks=py:fmt --bless`) and it also reformatted some code in subtrees (e.g. `clippy` or `rustc_codegen_gcc`) - I'm not sure how to handle that.

### CODE REVIEW

File: src/tools/tidy/config/black.toml
Comment (jieyouxu) [inline src/tools/tidy/config/black.toml:1]: Question: is there a way for `ruff` to also do ignores like this? AFAICT moving from `black` -> `ruff` in this case also drops this ignore list with no exceptions.

Comment (Kobzol) [reply src/tools/tidy/config/black.toml:1]: ruff already uses the exact same ignore list [here](https://github.com/rust-lang/rust/blob/71fe24824006ba9ffc54f8e166c53a90721ecebb/src/tools/tidy/config/ruff.toml).

Comment (jieyouxu) [reply src/tools/tidy/config/black.toml:1]: Ah nevermind, cool.

**Comment by jieyouxu** [comment]
I think trimming the python deps (and also making py fmt slightly faster) is nice, will let Onur assess if the improvement is worth the churn of reformatting

**Review by onur-ozkan — APPROVED** [review]
> If we decide to merge this, we'll need to "reformat the world" 

I liked the changes in this PR, how much change do you mean with "world"? Can you make a draft PR to show that?

**Comment by Kobzol** [comment]
I committed the change. It also modifies subtress though, e.g. `rustc_codegen_gcc`. We should probably add these to the ignore list in `ruff.toml`.

**Comment by onur-ozkan** [comment]
> We should probably add these to the ignore list in ruff.toml.

That or formatting them there first (would be pretty exhausting to file PR one by one).

--

I think the overall change is not too much and the changes are all in python files, so they likely won't cause so many conflicts with other ongoing tasks/PRs. We can land this easily once subtrees are handled.

File: src/tools/miri/test-cargo-miri/run-test.py
Comment (RalfJung) [inline src/tools/miri/test-cargo-miri/run-test.py:1]: Chances are if we ever edit this file in Miri it'll be formatted wrong... I know nothing about Python formatting.^^

But usually we exclude subprojects from formatting (e.g. via rustfmt).

**Comment by Kobzol** [comment]
Ok, I ignored `miri`, `clippy` and `rustc_codegen_gcc` and ran formatting again.

**Review by onur-ozkan — APPROVED** [review]

**Pull request merged**